### DISCUSSION TITLE
Initial authentication method : EndEntityCertificate

### DISCUSSION THREAD

**Discussion by pragya1998hal** [Keyfactor/ejbca-ce#753] [General]
Hello sir

I am using CMP configuration:

<img width="482" alt="image" src="https://github.com/user-attachments/assets/5f55d621-c05b-4c24-b19f-4d17f9e8c084" />

In the Initial certificate creation request I passed one End Entity certificate which is:

<img width="382" alt="image" src="https://github.com/user-attachments/assets/db150ee6-32c0-42d9-b2bc-eba01787fd9e" />

Do I need to pass same private key of End Entity certificate which I passed in extracerts field in certificate creation request?
I make one more EndEntity:

<img width="427" alt="image" src="https://github.com/user-attachments/assets/2bd08d90-d2e9-4e3e-a13a-158d6702ec6c" />

Is it possible that CA of End Entity certificate which I passed in extracerts field is different and CA for the certificate creation request will be different.
3.I have one End Entity certificate which I passed in extracerts field in certificate creation request. Then how I generate server and client certificated with two different CA?

Sir please guide me

**Comment by primetomas** [comment]
You have to work from the documentation, https://docs.keyfactor.com/ejbca/latest/cmp#id-(9.1.1)CMP-ClientModeforCMPClient_mode_for_CMP
There is also the source code available to figure out how it works internally.

You should try and see what error messages you get, messages returned in the server.log are usually very helpful to say what is needed and what is missing.

**Comment by pragya1998hal** [comment]
Hello sir,

As in CMP-Alias:

<img width="769" alt="image" src="https://github.com/user-attachments/assets/e6b0bf87-84f1-462d-bee3-f9e13ce4ca93" />


Under Default CA section we are only able to select one CA. As per my understanding The CA which I selected here is only used to sign certificate response. So Do I need to create two different CMP-Alias for client and server certificate separately?

**Reply by primetomas** [reply]
Default CA is only used for the Confirm message as the help text says. I recommend to not use Confirm at all but to use implicitConfirm instead. https://datatracker.ietf.org/doc/html/rfc4210#section-5.1.1.1

**Comment by pragya1998hal** [comment]
Hello sir please guide me
